src/fsm.test.ts

```diff
@@ -60,57 +63,95 @@ test('Invalid state changes', () => {
-  // `waitForChange` can take multiple arguments, for a "one of" check
-  await expect(
-    state.waitForChange('stopped', 'started')
-  ).resolves.toBeUndefined()
-
-  state.change('starting')
-  await expect(state.waitForChange('starting')).resolves.toBeUndefined()
-  await expect(
-    state.waitForChange('starting', 'stopping')
-  ).resolves.toBeUndefined()
+    for (const newState of ['stopped', 'stopping', 'started'] as State[]) {
+      expect(state.canChange(newState)).toBeFalsy()
+      expect(state.maybeChange(newState)).toBeFalsy()
+      expect(state.is('stopped')).toBeTruthy()
+    }
+  })
 })
 
-test('Await eventual state change', async () => {
-  expect.assertions(2)
-  const state = stateMachine()
+describe('waitForChange', () => {
+  test('Await immediate state change', async () => {
+    expect.assertions(4)
+    const state = stateMachine()
+    await expect(state.waitForChange('stopped')).resolves.toBeUndefined()
+    // `waitForChange` can take multiple arguments, for a "one of" check
+    await expect(
+      state.waitForChange('stopped', 'started')
+    ).resolves.toBeUndefined()
+
+    state.change('starting')
+    await expect(state.waitForChange('starting')).resolves.toBeUndefined()
+    await expect(
+      state.waitForChange('starting', 'stopping')
+    ).resolves.toBeUndefined()
+  })
+
+  test('Await eventual state change', async () => {
+    expect.assertions(2)
+    const state = stateMachine()
+
+    global.setTimeout(() => state.change('starting'), 250)
+    await expect(state.waitForChange('starting')).resolves.toBeUndefined()
+    global.setTimeout(() => state.change('started'), 250)
+    await expect(state.waitForChange('started')).resolves.toBeUndefined()
+  })
 
-  setTimeout(() => state.change('starting'), 250)
-  await expect(state.waitForChange('starting')).resolves.toBeUndefined()
-  setTimeout(() => state.change('started'), 250)
-  await expect(state.waitForChange('started')).resolves.toBeUndefined()
+  test('Eventual state change times out', async () => {
+    expect.assertions(1)
+    const state = stateMachine({ timeout: 250 })
+
+    await expect(state.waitForChange('starting')).rejects.toThrow(
+      'Did not complete in 250 ms'
+    )
+  })
 })
 
-test('Eventual state change times out', async () => {
-  expect.assertions(1)
-  const state = stateMachine({ timeout: 250 })
+describe('getElapsedTime', () => {
+  test('should work with initial state', async () => {
+    const state = stateMachine()
+    await setTimeout(1000)
+    expect(state.getElapsedTime()).toBeGreaterThanOrEqual(1000)
+  })
 
-  await expect(state.waitForChange('starting')).rejects.toThrow(
-    'Did not complete in 250 ms'
-  )
+  test('should work after state change', async () => {
+    const state = stateMachine()
+    await setTimeout(1000)
+    state.change('starting')
+    await setTimeout(1000)
+    expect(state.getElapsedTime()).toBeGreaterThanOrEqual(1000)
```

Comment:
It looks like the outcome here would be the same as if the first `await setTimeout(1000)` (line 134) weren't there.

If the idea is that the timer shouldn't start until we change to the "starting" state, we could test that by doing something like this:

```typescript
const state = stateMachine()
await setTimeout(5000) // Shouldn't affect elapsed time
state.change('starting')
await setTimeout(1000)
const elapsedTime = state.getElapsedTime()
expect(elapsedTime).toBeGreaterThanOrEqual(1000)
expect(elapsedTime).toBeLessThan(5000)
```

Reply:
The idea is that the timer is reset when you change state. The amount should be >=1000 && <2000. I added another `expect` call to the test.